pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -1079,9 +1131,14 @@ public void setLoggerFile(String loggerFile) {
   public String getUrl() {
     StringBuilder url = new StringBuilder(100);
     url.append("jdbc:postgresql://");
-    url.append(serverName);
-    if (portNumber != 0) {
-      url.append(":").append(portNumber);
+    for (int i = 0; i < serverNames.length; i++) {
+      if (i > 0) {
+        url.append(",");
+      }
+      url.append(serverNames[i]);
+      if (portNumbers.length >= i && portNumbers[i] != 0) {
```

Comment:
This assume `portNumbers` is not null (which it can be if the above changes are made).

Reply:
@teicher please check for null here.